server/neptune/workflows/internal/deploy/revision/queue/worker.go

```diff
@@ -105,10 +133,35 @@ func (w *Worker) Work(ctx workflow.Context) {
 	}
 }
 
+func (w *Worker) isValidRevision(ctx workflow.Context, msg terraform.DeploymentInfo, resp activities.CompareCommitResponse) bool {
```

Comment:
I mentioned this before when you had the struct. Returning a bool doesn't help anyone here, samra is going to have to go in and change the function signature when she's adding support for force applies so it's good to keep the future in mind when you're writing code.